src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -97,16 +135,99 @@ public AsyncJsonTokenParser(
     JsonToken jsonToken;
     try {
       jsonToken = deserialize(tokenString);
-    } catch (Exception e) {
+    } catch (InvalidJsonTokenException e) {
       return Futures.immediateFailedFuture(e);
     }
 
     return Futures.transform(verify(jsonToken), unused -> jsonToken, executor);
   }
 
+  /**
+   * Decodes the JWT token string into a JsonToken object. Does not perform
+   * any validation of headers or claims.
+   * Identical to {@link AbstractJsonTokenParser#abstractDeserialize(String)},
+   * except exceptions are caught and rethrown as {@link InvalidJsonTokenException}.
+   *
+   * @param tokenString The original encoded representation of a JWT
+   * @return Unverified contents of the JWT as a JsonToken
+   * @throws InvalidJsonTokenException with {@link ErrorCode#MALFORMED_TOKEN_STRING}
+   *   if the tokenString is not a properly formatted JWT.
+   */
+  public JsonToken deserialize(String tokenString) throws InvalidJsonTokenException {
+    try {
+      return abstractDeserialize(tokenString);
+    } catch (RuntimeException e) {
+      rethrowException(e);
+      throw new IllegalStateException("This should never be reached.");
+    }
```

Comment:
The reason I mapped only SignatureException and RuntimeException in the synchronous functions is that those exceptions are the only cases that we can guarantee that only InvalidJsonTokenException or RuntimeException is thrown (so the function signatures can just have throws InvalidJsonTokenException). 

The issue with having a universal rethrowing/mapping function is that we cannot guarantee that the only checked exception thrown will only be InvalidJsonTokenException for all exceptions. In the scenario that a checked exception cannot be mapped (there is no cause or the cause is not our custom exception), it needs to somehow be dealt with. Either, we ignore it, throw a runtime exception in its place, or change the function signature of the universal mapping function to have throws Exception. The first solution is deceiving, since we are covering up an exception that is probably important and the function is no longer universal for all exceptions. The second solution might work, except it becomes error prone due to a different exception thrown. The third solution works for the asynchronous case, since async functions catch all exceptions anyways. However, for the synchronous case, we can no longer have a one-liner or would have to change the function signature of all the synchronous functions to throws Exception and not just InvalidJsonTokenException.

Reply:
I changed it to be somewhat of a one-liner. However, unlike mapException (https://github.com/google/jsontoken/blob/ca922c1514e4d540257857874c746fdd3dac7532/src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java#L264), the function that I used for the synchronous one-liners mapExceptionAndThrow (https://github.com/google/jsontoken/blob/ca922c1514e4d540257857874c746fdd3dac7532/src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java#L291) does not work for all exceptions (https://github.com/google/jsontoken/blob/ca922c1514e4d540257857874c746fdd3dac7532/src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java#L288).

Reply:
I think that's fine, if someone adds a new checked exception they should also update the mapping function.